test/test_jacobi_theta.hpp

```diff
@@ -0,0 +1,517 @@
+            RealType(2) * jacobi_theta1(z, q) * jacobi_theta2(z, q) * jacobi_theta3(z, q) * jacobi_theta4(z, q),
+            eps);
+}
+
+// DLMF 20.7(iv) Transformations of Nome
+template <typename RealType>
+inline void test_transformations_of_nome(RealType z, RealType q, RealType eps) {
+    using namespace boost::math;
+
+    _check_close( // DLMF 20.7.11
+            jacobi_theta1(z, q) * jacobi_theta2(z, q) * jacobi_theta4(z + z, q * q),
+            jacobi_theta3(z, q) * jacobi_theta4(z, q) * jacobi_theta1(z + z, q * q),
+            eps);
+
+    _check_close( // DLMF 20.7.12
+            jacobi_theta1(z, q * q) * jacobi_theta4(z, q * q) * jacobi_theta2(z, q),
+            jacobi_theta2(z, q * q) * jacobi_theta3(z, q * q) * jacobi_theta1(z, q),
+            eps);
+}
+
+// DLMF 20.7(v) Watson's Identities
+template <typename RealType>
+inline void test_watsons_identities(RealType z, RealType w, RealType q, RealType eps) {
+    using namespace boost::math;
+
+    // Rearrange DLMF equations to get q*q on each side of the equality
+
+    _check_close( // DLMF 20.7.13
+            jacobi_theta1(z, q) * jacobi_theta1(w, q)
+            + jacobi_theta2(z + w, q * q) * jacobi_theta3(z - w, q * q),
+            jacobi_theta3(z + w, q * q) * jacobi_theta2(z - w, q * q),
+            eps);
+
+    _check_close( // DLMF 20.7.14
+            jacobi_theta3(z, q) * jacobi_theta3(w, q)
+            - jacobi_theta2(z + w, q * q) * jacobi_theta2(z - w, q * q),
+            jacobi_theta3(z + w, q * q) * jacobi_theta3(z - w, q * q),
+            eps);
+
+    _check_close( // MathWorld Eqn. 48
+            jacobi_theta3(z, q) - jacobi_theta2(z + z, q*q*q*q),
+            jacobi_theta3(z + z, q*q*q*q),
+            eps);
+
+    _check_close( // MathWorld Eqn. 49
+            jacobi_theta4(z, q) + jacobi_theta2(z + z, q*q*q*q),
+            jacobi_theta3(z + z, q*q*q*q),
+            eps);
+}
+
+template <typename RealType>
+inline void test_landen_transformations(RealType z, RealType tau, RealType eps) {
+    // A and B below are the reciprocals of their DLMF definitions
+    using namespace boost::math;
+
+    // DLMF 20.7.15 (reciprocal)
+    RealType A = jacobi_theta4tau(RealType(0), tau + tau);
+
+    _check_close( // DLMF 20.7.16
+            jacobi_theta1tau(z + z, tau + tau) * A,
+            jacobi_theta1tau(z, tau) * jacobi_theta2tau(z, tau),
+            eps);
+
+    _check_close( // DLMF 20.7.17
+            jacobi_theta2tau(z + z, tau + tau) * A,
+            jacobi_theta1tau(constants::quarter_pi<RealType>() - z, tau) * jacobi_theta1tau(constants::quarter_pi<RealType>() + z, tau),
+            eps);
+
+    _check_close( // DLMF 20.7.18
+            jacobi_theta3tau(z + z, tau + tau) * A,
+            jacobi_theta3tau(constants::quarter_pi<RealType>() - z, tau) * jacobi_theta3tau(constants::quarter_pi<RealType>() + z, tau),
+            eps);
+
+    _check_close( // DLMF 20.7.19
```

Comment:
If you want, you can use my "math_unit_test.hpp" and use CHECK_ULP_CLOSE if you think it'll be a bit more ergonomic. It also compiles much faster that Boost.Test.